Solution3.java

```diff
@@ -0,0 +1,37 @@
+// Time Complexity : O(N)
+// Space Complexity :
+// Did this code successfully run on Leetcode : Y
+// Any problem you faced while coding this :
```

Comment:
Good Job! Keep up the good work. Please state the space complexity of this solution.